### PR TITLE
bugfix:fix the NPE of RedisTransactionStoreManager when get branch tr…

### LINKED ISSUE #3525: seata1.4.0 启动报错
重启seata服务时，报错如下，导致无法启动
Caused by: java.lang.NullPointerException
	at java.util.Comparator.lambda$comparing$77a9974f$1(Comparator.java:469)
	at java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
	at java.util.TimSort.sort(TimSort.java:220)
	at java.util.Arrays.sort(Arrays.java:1512)
	at java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:353)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:483)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566)
	at io.seata.server.storage.redis.store.RedisTransactionStoreManager.readBranchSessionByXid(RedisTransactionStoreManager.java:463)
	at io.seata.server.storage.redis.store.RedisTransactionStoreManager.readSession(RedisTransactionStoreManager.java:318)
	at io.seata.server.storage.redis.store.RedisTransactionStoreManager.lambda$readSession$2(RedisTransactionStoreManager.java:357)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)

### PR DESCRIPTION
**Pull request by tanggen**
### Ⅰ. Describe what this PR did
fix the NEP of RedisTransactionStoreManager when get branch transactions

### Ⅱ. Does this pull request fix one issue?

fixes #3525

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
this NEP is not happening every time，it happens when submit a lot transaction，then sometime async_committer delete brach map in redis befor global transaction read it. 

### Ⅳ. Describe how to verify it
submit a lot transaction，break point at BranchTransactionDO.compareTo() with condition:(this.getGmtCreate() == null || branchTransactionDO == null || branchTransactionDO.getGmtCreate() == null) ,maybe u will get the NEP.

### Ⅴ. Special notes for reviews
the NEP happened many times in testing with more than 100W transaction，after fixed，never happend in 100W transaction .

### CODE REVIEW

**Comment by tanggen** [comment]
fix input import java.util.*

**Review by a364176773 — APPROVED** [review]
LGTM

**Comment by jsbxyyx** [comment]
please git config relation github

File: changes/en-us/1.5.0.md [hunk 1]
```diff
@@ -43,6 +43,7 @@
   - [[#3451](https://github.com/seata/seata/pull/3451)] fix set auto-commit to true when local transactions are not being used. Failure to compete for a lock causes the global transaction to exit, invaliding the global row lock and dirty writing of the data.
   - [[#3481](https://github.com/seata/seata/pull/3481)] fix seata node refresh failure because consul crash
   - [[#3491](https://github.com/seata/seata/pull/3491)] fix typo in README.md
+  - [[#3531](https://github.com/seata/seata/pull/3531)] fix the NEP of RedisTransactionStoreManager when get branch transactions
```
Comment (caohdgege) [inline changes/en-us/1.5.0.md:46]: NEP? NPE?

Comment (tanggen) [reply changes/en-us/1.5.0.md:46]: sorry，my bad! should be NPE

**Review by caohdgege — APPROVED** [review]
LGTM

**Review by ph3636 — APPROVED** [review]
LGTM

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**